flink-runtime/src/main/java/org/apache/flink/runtime/state/filesystem/FsStateBackend.java

```diff
@@ -97,6 +100,27 @@ public FsStateBackend(String checkpointDataUri) throws IOException {
 	 *
 	 * @param checkpointDataUri The URI describing the filesystem (scheme and optionally authority),
 	 *                          and the path to the checkpoint data directory.
+	 * @param asynchronousSnapshots Switch to enable asynchronous snapshots.
+	 *
+	 * @throws IOException Thrown, if no file system can be found for the scheme in the URI.
+	 */
+	public FsStateBackend(String checkpointDataUri, boolean asynchronousSnapshots) throws IOException {
```

Comment:
For now, I suggest to keep it as is and as soon as your PR goes in, we have a followup, that introduced a single entry point to a builder for all backend related configuration options that helps the user to see and navigate all options in a more centralized way, e.g. `KeyedBackends.heap().async(true).create()`.